compiler/assemble.js

```diff
@@ -47,7 +47,48 @@ const encodeNames = funcs => {
   ];
 };
 
-export default (funcs, globals, tags, pages, data, flags, noTreeshake = false) => {
+const encodeDebugInfo = (funcs, globals, tags, exceptions, pages, data, excMode) => {
+  const encodeSection = (id, section) => [
+    id,
+    ...unsignedLEB128(section.length),
+    ...section
+  ];
+  let exceptionSection = [];
+  switch (excMode) {
+    case 'lut':
+      exceptionSection.push(0);
+      unsignedLEB128_into(exceptions.length, exceptionSection);
+      for (let i = 0; i < exceptions.length; i++) {
+        unsignedLEB128_into(i, exceptionSection);
+        exceptionSection.push(...encodeString(exceptions[i].constructor ?? ""));
+        exceptionSection.push(...encodeString(exceptions[i].message));
+      }
+      break;
+    case 'stack':
+      exceptionSection.push(1);
+      break;
+    case 'stackest':
+      exceptionSection.push(2);
+      break;
+    case 'partial':
+      exceptionSection.push(3);
+      unsignedLEB128_into(exceptions.length, exceptionSection);
+      for (let i = 0; i < exceptions.length; i++) {
+        unsignedLEB128_into(i, exceptionSection);
+        exceptionSection.push(...encodeString(exceptions[i].constructor ?? ""));
+      }
+      break;
+  }
+  const typeSection = [ valtypeBinary ];
+
+  // TODO: add more information like where funcrefs are, etc.
+  return [
+    ...encodeSection(0, exceptionSection),
+    ...encodeSection(1, typeSection),
+  ];
+};
+
+export default (funcs, globals, tags, exceptions, pages, data, flags, noTreeshake = false) => {
```

Comment:
This change is unneeded now that stack is used by default?

Reply:
Yeah, sure, maybe keep a debug section for other stuff though

Reply:
Okay, removed that exception table debug information